adagrams/game.py

```diff
@@ -1,13 +1,133 @@
 from random import randint
 
+#adding the letter pool to game.py for reference
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+#including score counter 
+LETTER_SCORE = {
+    1:['A','E','I','O','U','L','N','R','S','T'],
+    2:['D','G'],
+    3:['B','C','M','P'],
+    4:['F','H','V','W','Y'],
+    5:['K'],
+    8:['J','X'],
+    10:['Q','Z']
+}
+
+#function for converting the letter pool in a copied list for use
+def available_letters():
+
+    available_letters = []
+
+    for letter, quantity in LETTER_POOL.items():
+        available_letters.extend(([letter] * quantity))
+
+    return available_letters
+    
+
 def draw_letters():
-    pass
+
+    letters_list = available_letters()
+    letters = []
+    
+    while len(letters) < 10:
+        i = randint(1, len(letters_list)-1)
+        letters.append(letters_list[i])
+        letters_list.pop(i)
```

Comment:
The list `pop()` builtin method has two different versions with slightly different implementations. `pop()` without any parameters will simply remove the item at the end of the list. This happens in O(1) time. `pop(index)` with a parameter removes the value at the index parameter and then shifts everything that comes after it down one index in the list. This ends up happening in O(n) time. While there are situations where we will need to use the parameterized version, if we can use the version with no parameters, that is ideal. 

One way we could do this is to actually swap the value at the random index with the value at the back of the list and then remove it from the end with `pop()`. While it does add in an extra step, it does decrease the complexity! This might look something like the following:

```suggestion
        letters_list[i], letters_list[-1] = letters_list[-1], letters_list[i] # Note we can easily swap values using this dual assignment.
        letters_list.pop()
```